src/mavsdk/core/mavlink_ftp_client.cpp

```diff
@@ -1210,8 +1220,35 @@ void MavlinkFtpClient::timeout()
                 }
 
                 {
-                    start_timer();
-                    send_mavlink_ftp_message(work->payload);
+                    // This happens when we missed the last ack containing burst complete.
+                    // We have already a file size, so we don't need to start at the
+                    // beginning any more.
+                    if (item.file_size != 0 && item.current_offset != 0) {
+                        // In that case start requesting what we missed.
+                        if (item.current_offset == item.file_size && item.missing_data.empty()) {
+                            // We are done anyway.
+                            item.callback(ClientResult::Success, {});
+                            download_burst_end(*work);
+                            work_queue_guard.pop_front();
+                        } else {
+                            // The burst is supposedly complete but we still need data because
+                            // we missed some, so request next without burst.
+                            // We presumably missed the very last chunk.
+                            if (item.current_offset < item.file_size) {
+                                item.missing_data.emplace_back(DownloadBurstItem::MissingData{
```

Comment:
Maybe we should start a new burst download because if we miss the first burst ack, we would download all the file with standard read.